qutip/qobj.py

```diff
@@ -1843,7 +1843,7 @@ def extract_states(self, states_inds, normalize=False):
 
         """
         if self.isoper:
-            q = Qobj(self.data[states_inds, :][:, states_inds])
+            q = Qobj(self.data[states_inds, :][:, states_inds], isherm=self._isherm)
```

Comment:
A non-hermitian could become hermitian,
```suggestion
            q = Qobj(self.data[states_inds, :][:, states_inds], isherm=self._isherm or None)
```